buildSrc/src/main/kotlin/kotest-publishing-conventions.gradle.kts

```
@@ -0,0 +1,87 @@
+import org.gradle.api.publish.PublicationContainer
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.kotlin.dsl.get
+import org.gradle.kotlin.dsl.invoke
+import org.gradle.kotlin.dsl.provideDelegate
+import org.gradle.kotlin.dsl.register
+import org.gradle.kotlin.dsl.repositories
+import org.gradle.kotlin.dsl.signing
+
+plugins {
+   signing
+   `java-library`
+   `maven-publish`
+}
+
+group = "io.kotest.extensions"
+version = Ci.version
+
+val publications: PublicationContainer = (extensions.getByName("publishing") as PublishingExtension).publications
+
+val ossrhUsername: String by project
+val ossrhPassword: String by project
+val signingKey: String? by project
+val signingPassword: String? by project
+
+signing {
+   useGpgCmd()
+   if (signingKey != null && signingPassword != null) {
+      useInMemoryPgpKeys(signingKey, signingPassword)
+   }
+   if (Ci.isRelease) {
+      sign(publications)
+   }
+}
+
+java {
+   withJavadocJar()
+   withSourcesJar()
+}
+
+publishing {
+   repositories {
+      maven {
+         val releasesRepoUrl = uri("https://oss.sonatype.org/service/local/staging/deploy/maven2/")
+         val snapshotsRepoUrl = uri("https://oss.sonatype.org/content/repositories/snapshots/")
```

```suggestion
         val releasesRepoUrl = uri("https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/")
         val snapshotsRepoUrl = uri("https://s01.oss.sonatype.org/content/repositories/snapshots/")
```

Sonatype changed the URL for publishing